### PR TITLE
Trace read requests

### PR DESCRIPTION
**Pull request by bojanserafimov**
Add tenant option to record all read requests and write the trace to a file. The intended use case is to run a test, record a trace, and then either:
- inspect the trace for bugs (duplicate requests, etc)
- visualize the trace
- replay the trace in various ways (serial, parallel, real-time, etc)

I write the traces in binary format to save space. They can be big.

### CODE REVIEW

File: pageserver/src/config.rs [hunk 1]
```diff
@@ -414,6 +414,22 @@ impl PageServerConf {
...
 
+    pub fn traces_path(&self) -> PathBuf {
+        self.workdir.join("traces")
+    }
+
+    pub fn trace_path(
+        &self,
+        tenant_id: &TenantId,
+        timeline_id: &TimelineId,
+        connection_id: &TimelineId, // TODO make a new type
```
Comment (bojanserafimov) [inline pageserver/src/config.rs:425]: will address this before merging

File: test_runner/performance/test_trace_replay.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    tenant, _ = env.neon_cli.create_tenant(
+        conf={
+            "trace_read_requests": "true",
+        }
+    )
+
+    # TODO This doesn't work because I haven't updated tenant_config_handler
+    # env.neon_cli.config_tenant(tenant, conf={
+    #     "trace_read_requests": "true",
+    # })
```
Comment (bojanserafimov) [inline test_runner/performance/test_trace_replay.py:21]: will address this before merging

File: test_runner/performance/test_trace_replay.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from contextlib import closing
+
+from fixtures.neon_fixtures import NeonEnvBuilder
+
+
+# This test demonstrates how to collect a read trace. It's useful until
+# it gets replaced by a test that actually does stuff with the trace.
+def test_read_request_tracing(neon_env_builder: NeonEnvBuilder):
```
Comment (knizhnik) [inline test_runner/performance/test_trace_replay.py:8]: Test is called trace replay, bit it seems to be not replaying any traces.

**Review by knizhnik — APPROVED** [review]
As far as trace format is binary we definitely need some utility for dumping traces.

**Pull request merged**